### PR TITLE
fix: update dependencies and asset lock proof handling

### PR DESCRIPTION
**Pull request by pauldelucia**

### CODE REVIEW

File: src/backend/strategies.rs [hunk 1]
```diff
@@ -661,6 +661,51 @@ pub async fn run_strategy_task<'s>(
...
+                                    },
+                                    Err(e) => tracing::error!("Error broadcasting and retrieving asset lock: {:?}", e),
+                                }
+                            },
+                            Err(e) => {
+                                tracing::error!("Error creating asset lock transaction: {:?}", e)
+                            },
+                        }
+                    } else {
+                        tracing::error!("Wallet not loaded");
```
Comment (lklimek) [inline src/backend/strategies.rs:703]: Maybe we should break execution here? (return or maybe panic)?

File: src/backend/strategies.rs [hunk 2]
```diff
@@ -661,6 +661,51 @@ pub async fn run_strategy_task<'s>(
...
+                                    Ok(asset_lock_proof) => {
+                                        tracing::info!("Got {} asset lock proofs", i+1);
+                                        num_asset_lock_proofs_obtained += 1;
+                                        asset_lock_proofs.push((asset_lock_proof, asset_lock_proof_private_key));
+                                    },
+                                    Err(e) => tracing::error!("Error broadcasting and retrieving asset lock: {:?}", e),
+                                }
+                            },
+                            Err(e) => {
+                                tracing::error!("Error creating asset lock transaction: {:?}", e)
```
Comment (lklimek) [inline src/backend/strategies.rs:699]: Maybe we should break execution here? (return or maybe panic)?

File: src/backend/strategies.rs (hunk 1 above)
Comment (pauldelucia) [reply src/backend/strategies.rs:703]: Yes. Fixed.

File: src/backend/strategies.rs (hunk 2 above)
Comment (pauldelucia) [reply src/backend/strategies.rs:699]: I'm not sure it's necessary here

**Review by lklimek — APPROVED** [review]

**Pull request merged**